sbi/utils/pyroutils.py

```diff
@@ -33,3 +40,142 @@ def prior():
             transforms[name] = biject_to(fn.support).inv
 
     return transforms
+
+
+class EstimatorDistribution(pyro.distributions.TorchDistribution):
+    """
+    Base class for a conditioned `sbi` estimator wrapped as a Pyro distribution.
+    """
+
+    def __init__(self, estimator: Any, condition: torch.Tensor):
+        self._estimator = estimator
+        condition_shape, event_shape = self.get_condition_and_event_shapes()
+        self._check_condition_shape(condition)
+        self._condition = condition
+        self._condition_reshaped = condition.reshape(-1, *condition_shape)
```

Comment:
note for me: this add the `sample_dim` to `condition` because this is was pyro needs. another scenario where having `sample_dim` for `condition` as well throughout `sbi` would be good.

Reply:
Ah, it's the other way around. Pyro is fine with `condition` having shape `(*batch_shape, *condition_shape)`, but SBI (at least for `ConditionalDensityEstimator`) wants it to have the shape `(batch_dim, *condition_shape)`. So here we flatten the batch shape (if present) into a single batch dimension or add a singleton dimension if the batch shape is empty.